app/services/chunkers.rb

```diff
@@ -0,0 +1,35 @@
+module Chunkers
+  class UnknownChunkingMethod < StandardError; end
+
+  include Enumerable
+
+  module InputType
+    PLAIN_TEXT = 0
+    COMMON_MARK = 1
+  end
+
+  CHUNKING_METHODS = [
+    { id: :recursive_split, name: "Recursive Split" },
+    { id: :basic, name: "Basic" },
+  ].freeze
+
+  def self.chunker_for(chunking_profile)
+    chunking_method = chunking_profile.method.to_sym
+    if chunking_method == :bytes
+      # Mapping :bytes method for backwards compatibility since
+      # the DB will have existing chunking profiles that we might want to "reprocess"
+      Rails.logger.info("Backwards compatibility - mapping :bytes to :recursive_split chuking method")
+      chunking_method = :recursive_split
```

Comment:
Is :bytes not more like :basic?

Reply:
Yes and no only because previously we used the `bytes` method and let the parser pick _actual_ the chunking method. This means that prior to this PR, PDF chunking used the basic chunker but the other formats used the recursive split chunker. However with this separation of chunker from parser, I had to pick one or other for backwards compatibility, and I went with the "better" of the two strategies. 

What we could do, ideally, is prompt the user on a "re-process" request if they want to keep the previous profile or change it. 

With that said, I am OK with `bytes` mapping to either; if we have a lot more PDFs in deployed systems, then using `basic` is gonna provide most compatibility. Let me know. Easy fix.